packages/relay/src/lib/eth.ts

```diff
@@ -119,16 +132,28 @@ export class EthImpl implements Eth {
 
     //contractTransactionGas is in tinycents * 1000, so the final multiplier is truncated by 3 zeroes for
     // the conversion to weibars
-    return Math.ceil(contractTransactionGas / exchangeRates.currentRate.cents *
-      exchangeRates.currentRate.hbars * 10_000_000);
+    return Math.ceil(
+      (contractTransactionGas / exchangeRates.currentRate.cents) *
+        exchangeRates.currentRate.hbars *
+        10_000_000
+    );
   }
 
   /**
    * Gets the most recent block number.
    */
-  async blockNumber() {
+  async blockNumber(): Promise<number> {
     this.logger.trace('blockNumber()');
-    return await this.mirrorNode.getMostRecentBlockNumber();
+    const blockResponse = await this.mirrorNodeClient.getLatestBlock();
```

Comment:
Here's another way that also handles errors.
IF the mirror node doesn't find it a valid 404 with a valid response will be returned so I removed the block undefined check
```ts
return this.mirrorNodeClient.getLatestBlock()
      .then(block => {
        if (block.number === undefined) {
          this.logger.debug('Failed to retrieve latest block');
          // block not found
          return null;
        }

        this.logger.trace('blockNumber=' + block.number);
        return block.number;
      })
      .catch((e: any) => {
        this.logger.error(e, 'Error encountered retrieving latest block');
        return null;
      });
```